src/core/Global.ttslua

```diff
@@ -456,358 +268,281 @@ function drawChaostoken(params)
   end
 end
 
+---------------------------------------------------------
+-- token spawning
+---------------------------------------------------------
+
 function spawnToken(params)
-  -- Position to spawn,
-  -- rotation vector to apply
-  -- translation vector to apply
-  -- token type
   local position = params[1]
   local tokenType = params[2]
+  local rotation = params[3] or {0, 270, 0}
   local tokenData = TOKEN_DATA[tokenType]
-  if tokenData == nil then
-    error("no token data found for '" .. tokenType .. "'")
-  end
+  if tokenData == nil then error("no token data found for '" .. tokenType .. "'") end
 
   local token = spawnObject({
     type = 'Custom_Token',
     position = position,
-	rotation = {x=0, y=270, z=0}
+	  rotation = rotation
   })
   token.setCustomObject({
     image = tokenData['image'],
     thickness = 0.3,
-    merge_distance = 5.0,
-    stackable = true,
+    merge_distance = 5,
+    stackable = true
   })
-  token.use_snap_points=false
+  token.use_snap_points = false
   token.scale(tokenData['scale'])
   return token
 end
 
-function round(params) -- Parameter (int number, int numberDecimalPlaces)
-  return tonumber(string.format("%." .. (params[2] or 0) .. "f", params[1]))
-end
+---------------------------------------------------------
+-- chaos token stat tracker
+---------------------------------------------------------
 
-function roundposition(params) -- Parameter (Table position)
-  return {round({params[1], 2}),round({params[2], 2}),round({params[3], 2})}
+function take_callback(object_spawned, mat)
+  local player = mat.getGUID()
+  local image = object_spawned.getCustomObject().image
+  PULLS[image]                = (PULLS[image] or 0) + 1
+  PLAYER_PULLS[player][image] = (PLAYER_PULLS[player][image] or 0) + 1
 end
 
-function isEqual(params) --Parameter (Table table1, Table table2) returns true if the tables are equal
-  if params[1][1] == params[2][1] and params[1][2] == params[2][2] and params[1][3] == params[2][3] then
-    return true
+function printOrResetStats(_, _, isRightClick)
```

Comment:
Let's split this up for clarity:
function handleStatTrackerClick(_, _, isRightClick)
  if isRightClick then
    resetStatTracker()
  else
    printStats()
  end
end

Associated functions would take the blocks from the current if